src/plugins/navsat_map/NavSatMap.qml

```diff
@@ -0,0 +1,167 @@
+
+Item {
+  id: navSatMap
+  property double latitude: 0.0
+  property double longitude: 0.0
+  property bool centering: true
+
+  Layout.minimumWidth: 280
+  Layout.minimumHeight: 455
+  anchors.topMargin: 5
+  anchors.leftMargin: 5
+  anchors.rightMargin: 5
+  anchors.fill: parent
+
+  ColumnLayout {
+    id: configColumn
+    width: parent.width
+
+    RowLayout {
+      width: parent.width
+
+      Layout.fillWidth: true
+      Layout.fillHeight: true
+
+      RoundButton {
+        text: "\u21bb"
+        Material.background: Material.primary
+        onClicked: {
+          NavSatMap.OnRefresh();
+        }
+      }
+
+      ComboBox {
+        id: combo
+        Layout.fillWidth: true
+        model: NavSatMap.topicList
+        onCurrentIndexChanged: {
+          if (currentIndex < 0) {
+            return;
+          }
+
+          NavSatMap.OnTopic(textAt(currentIndex));
+        }
+        ToolTip.visible: hovered
+        ToolTip.delay: Qt.styleHints.mousePressAndHoldInterval
+        ToolTip.text: qsTr("Ignition Transport topics publishing NavSat messages")
+      }
+    }
+
+    RowLayout {
+      width: parent.width
+      Item {
+        height: 1
+        Layout.fillWidth: true
+      }
+      Text {
+        text: "<b>Latitude: </b>" + latitude.toFixed(6)
+      }
+
+      Text {
+        text: "<b>Longitude: </b>" + longitude.toFixed(6)
+      }
+      Item {
+        height: 1
+        Layout.fillWidth: true
+      }
+    }
+  }
+
+  Plugin {
+    id: mapPlugin
+    name: "osm"
+  }
+
+  Map {
+    id: map
+    anchors.top: configColumn.bottom
+    anchors.bottom: parent.bottom
+    anchors.left: parent.left
+    anchors.right: parent.right
+
+    anchors.topMargin: 5
+    anchors.leftMargin: -5
+    anchors.rightMargin: -5
+
+    MapCircle {
+      id: circle
+      center {
+        latitude: latitude
+        longitude: longitude
+      }
+      radius: 23.0
+      color: '#4285f4'
+      border.width: 3
+      border.color: "white"
+    }
+
+    plugin: mapPlugin
+    center: centering ? QtPositioning.coordinate(latitude, longitude) : center
+    copyrightsVisible: false
+    zoomLevel: 16
+
+    gesture.onPanStarted: {
+      centering = false
+    }
+
+    onZoomLevelChanged: {
+      // Scaling the location marker according to zoom level.
+      // Marker radius = meters per pixel * 10
+      circle.radius = 156543.03392 * Math.cos(latitude * Math.PI / 180) / Math.pow(2, map.zoomLevel) * 10
+    }
+  }
+
+  RoundButton {
+    height: 40
+    width: 40
+    anchors.top: map.top
+    anchors.right: map.right
+
+    // Strangely, \u2316 doesn't work on Bionic
+    text: "\u2732"
+    font.pixelSize: 25
+
+    Material.background: Material.primary
+    Material.foreground: centering ? "white" : "black"
+
+    onClicked: {
+      centering = true
+    }
+
+    hoverEnabled: true
+
+    ToolTip.delay: Qt.styleHints.mousePressAndHoldInterval
+    ToolTip.visible: hovered
+    ToolTip.text: qsTr("Keep point on center")
```

Comment:
When I first read this before clicking, I didn't understand what this meant. How about something like, `"Reset to center"` or `"Reset point to center"`?

Reply:
Yeah I agree it's confusing. Ignition RViz [just says "Center"](https://github.com/ignitionrobotics/ign-rviz/blob/632a18ade4ece9d0602a10ae48559c8b6d8392c4/ign_rviz_plugins/res/qml/GPSDisplay.qml#L213), so I added the "Keep point on" here in an attempt to make it clearer, but I agree it's still confusing.

I think "Reset" may give the wrong impression though, because it's not a one-time thing. Maybe something like "Track point", "Track location", "Enable tracking"...?

Reply:
Ah I see, I like "Track point" or "Track position"

Reply:
Thanks, went with "Track position": c898d07